ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java

```diff
@@ -126,6 +126,11 @@ public ImportDeclarationSnippet createImportSnippet(Node node) {
     @Override
     public VariableDeclarationSnippet createVariableDeclarationSnippet(Node node) {
         ModuleVariableDeclarationNode dclnNode;
+        if (isContainsIsolated(node)) {
+            addErrorDiagnostic("Isolation not allowed in Ballerina shell");
+            return null;
```

Comment:
We are returning null here, and when consuming this API, we might miss the `null` check. Hence what if we introduce a new API which returns an `Optional?`

Reply:
Since BasicSnippetFactory class has many uses in the shell implementation this will be addressed as a separate task. 
issue : #35405